doc/arkode/guide/source/Usage/ASA.rst

```diff
@@ -0,0 +1,159 @@
+.. _ARKODE.Usage.ASA:
+
+Adjoint Sensitivity Analysis
+============================
+
+The previous sections discuss using ARKODE for the integration of forward ODE models. This section
+discusses how to use ARKODE for adjoint sensitivity analysis as introduced in
+:numref:`ARKODE.Mathematics.ASA`. To use ARKStep for adjoint sensitivity analysis (ASA), users
+simply setup the forward integration as usual (following :numref:`ARKODE.Usage.Skeleton`) with a few
+differences. Below we provide an updated version of the ARKode usage in section
+:numref:`ARKODE.Usage.Skeleton` where steps that are unchanged from are *italicized*. The example
+code ``examples/arkode/C_serial/ark_lotka_volterra_asa.c`` demonstrates these steps in detail.
+
+.. index:: Adjoint Sensitivity Analysis user main program
+
+#. *Initialize parallel or multi-threaded environment, if appropriate.*
+
+#. *Create the SUNDIALS simulation context object*
+
+#. *Set problem dimensions, etc.*
+
+#. *Set vector of initial values*
+
+#. *Create a stepper object*
+
+#. *Set optional inputs*
+
+#. *Specify rootfinding problem*
+
+#. Create a :c:type:`SUNAdjointCheckpointScheme` object
+
+   Create the :c:type:`SUNAdjointCheckpointScheme` object by calling ``SUNAdjointCheckpointScheme_Create_*``.
+   Available :c:type:`SUNAdjointCheckpointScheme` implementations are found in
+   section :numref:`SUNAdjoint.CheckpointScheme`.
+   
+#. Attach the checkpoint scheme object to ARKODE 
+   
+   Call :c:func:`ARKodeSetAdjointCheckpointScheme()`.
+
+#. *Advance solution in time*
+
+#. *Get optional outputs*
+
+#. Create the sensitivities vector with the terminal condition
+
+   The sensitivities vector must be an instance of the :ref:`ManyVector N_Vector implementation <NVectors.ManyVector>`.
+   You will have one subvector for the initial condition sensitivities and an additional subvector if you
+   want sensitivities with respect to parameters.
+   The vector should contain the terminal condition from the forward problem.
+
+#. Create the :c:type:`SUNAdjointStepper` object
+
+   Call :c:func:`ERKStepCreateAdjointStepper` or :c:func:`ARKStepCreateAdjointStepper`.
+
+#. Set either the Jacobian-vector product, vector-Jacobian product, or Jacobian evaluation functions
+
+   Users must supply one of:
+   
+   * :math:`(\partial f/\partial y)^*v`,
+   * :math:`v^*(\partial f/\partial y)`,
```

Comment:
Does the asterisk here represent matrix-vector or vector-matrix multiplication?  Or does this represent the complex conjugate transpose?  Given that `Mathematics.rst` used the non-complex transpose "^T", and SUNDIALS does not yet support complex arithmetic, then should this instead be the following?
```suggestion
   * :math:`(\partial f/\partial y)^Tv`,
   * :math:`v^T(\partial f/\partial y)`,
```

Reply:
It represents the complex conjugate transpose. I think we can change it back to `^T` for now as you suggested.

Reply:
We discussed using `^*` or `^T` at an earlier meeting and decided to go with `^*` to be consistent with the CVODES documentation and so this does not need to be updated when complex support is added